module_poll.lua

```diff
@@ -0,0 +1,496 @@
+-- Copyright (C) 2020 Antoine James Tournepiche & Jérôme Leclercq
+-- This file is part of the "Not a Bot" application
+-- For conditions of distribution and use, see copyright notice in LICENSE
+
+local client = Client
+local discordia = Discordia
+local bot = Bot
+local enums = discordia.enums
+
+-- global
+local polls
+
+Module.Name = "poll"
+
+-- TODO? Ajouter ça dans les utils ?
+function HasOneOfTheseRoles(member, roles)
+    for _, roleId in ipairs(roles) do
+        if member:hasRole(roleId) then
+            return true
+        end
+    end
+    return false
+end
+
+-- TODO Supprimer ça ou ajouter dans les utils ?
+function SendNotImplemented(channel)
+    channel:send("This feature is not yet implemented!")
+end
+
+function Module:CheckPermissions(member)
+    if member:hasPermission(enums.permission.administrator) then
+        return true
+    end
+    return HasOneOfTheseRoles(member, self:GetConfig(member.guild).AllowedRoles)
+end
+
+-- TODO? Ajouter option de cooldown entre 2 sondages pour un même membre
+-- TODO? (plus tard) Ajouter option pour changer la couleur (param par défaut et action 'color')
+function Module:GetConfigTable()
+	return {
+        {
+            Array = true,
+            Name = "AllowedRoles",
+            Description = "Roles allowed to create polls",
+            Type = bot.ConfigType.Role,
+            Default = {}
+        },
+        {
+            Array = true,
+            Name = "SpecifyChannelAllowedRoles",
+            Description = "Roles allowed to specify where to send a poll",
+            Type = bot.ConfigType.Role,
+            Default = {}
+        },
+        {
+            Name = "DefaultPollChannel",
+            Description = "Where should polls be sent if no channel is set on init",
+            Type = bot.ConfigType.Channel,
+            Optional = true
+        },
+        {
+            Name = "DefaultPollDuration",
+            Description = "Default poll duration if no duration is set on init",
+            Type = bot.ConfigType.Duration,
+            Default = 24 * 60 * 60
+        },
+        {
+            Name = "DeletePollOnExpiration",
+            Description = "Delete original poll message on expiration",
+            Type = bot.ConfigType.Boolean,
+            Default = true
+        }
+    }
+end
+
+function Module:OnReady()
+	self.Clock:start()
+end
+
+function Module:OnUnload()
+	if (self.Clock) then
+		self.Clock:stop()
+	end
+end
+
+function Module:OnLoaded()
+    polls = {}
```

Comment:
Don't use a global table for storing polls, use module data (or persistent data if required)